### ISSUE TITLE
client connects, but no sharing

### ISSUE THREAD

**Issue by timeweaver** [debauchee/barrier#575]
### Operating Systems ###

Server: RHEL 8

Client: Windows 10

### Barrier Version ###

2.3.2

### Steps to reproduce bug ###

My client connects, log on the client shows connected to server, log on server shows client connecting.  But mouse won't travel between screens.

### Other info ###

This is a first time setup - I disabled selinux and shut down firewalld on my redhat server in order to allow the client to connect.  Not seeing errors in either log.

**Comment by timeweaver** [comment]
Ok - I guess you can close this - It just magically started working - I have no idea why.  I disabled ssl on both and nothing changed, so I re-enabled it.  boom.  working.

**Comment by timeweaver** [comment]
And now it's not - not idea - help!!

**Comment by the-wes** [comment]
Can you provide some log output? Turn it up to debug1 and see if you get any messages when trying to switch screens? Attach a screenshot of your server config?

**Comment by timeweaver** [comment]
Ok, here's a log file with debug turned on.  The txt file is the saved config.  I don't see anything showing up when I move my mouse against the edge of the screen like it is even trying to switch.  I've had two occasions where it started working for about 10 minutes or so, then just stopped.  Sadly, I couldn't get it to work during this log capture. 

[barrier.log](https://github.com/debauchee/barrier/files/4283983/barrier.log)

[barrier.txt](https://github.com/debauchee/barrier/files/4284002/barrier.txt)

**Comment by nsclong** [comment]
I have the same issue. I am running Debian 10 (server) and Windows 10 (client), both on Barrier version 2.2.0.

The machines are connecting to each other fine and talking to each other. Debug1 level log attached. However mouse just will not share. It stops at the edges of the server screen.

When I move my mouse to the edge of the screen to attempt to move to the client machine, nothing changes in the log.

[barrier-server.log](https://github.com/debauchee/barrier/files/4297587/barrier-server.log)

**Comment by timeweaver** [comment]
Turns out my company won't let me use this software anyway.  Thanks for your efforts, I'm gonna go cry in my milk now.

**Comment by ik3umt** [comment]
Same issue here, server win10 + barrier 2.3.2 , client latest debian10 + barrier 2.2.0
Client connects to server, both ok on logs, logs tell entering and leaving screens at relative position, keyboard OK but no control over mouse

**Comment by nsclong** [comment]
OK. I have this fixed. I found that I had an old copy of 'Mouse Without Borders' on my Windows 10 machine. After uninstalling and rebooting, Barrier works beautifully with Debian server, Windows client. Thanks for all the developers for a great product.

**Comment by geertdobbels** [comment]
Seems to be the same as #589 where I posted, but in the meantime I read this thread and also captured the log file with debug1.

Server : Linux Fedora 31 (installed via flatpak)
Client: Windows 10
Barrier Version 2.3.2 on both computers.

I had barrier working on the same configuration (though I do not remember the barrier version number) for several months without problems but then due to a technical problem, I had to reinstall Fedora31, and since then I have the following issue:
Barrier server clearly sees the client, as can be seen in the logfile.  After changing the log level to Debug1, I only once could see a couple of messages related to leaving the screen on the bottom (which I even do not remember having done), but after that no more messages to that effect appear, no matter how and on which border I try to leave the server screen.
Server has dual screen, but I tried with single screen: no change.
I also tried with and without SSL, no difference.
[barrier.log](https://github.com/debauchee/barrier/files/4503874/barrier.log)

**Comment by geertdobbels** [comment]
After starting barrier from the command shell, I saw a message related to a missing wayland-egl plugin, which made me think that wayland might be the problem (wayland is on by default in gnome 3.4 or at least in fedora31).  This made me think that on my previous fedora31 install, I used to login under Xorg gnome, and not under wayland .....
So I logged out and logged back in but under xOrg gnome instead of the default wayland, and now barrier seems to be working.  The question now is how to make barrier working under wayland....

**Comment by scottajohnson** [comment]
I had the same problem with a Win7 Server and raspian clients. Scroll lock was turned on and that locked the mouse to the windows screens.

**Comment by deisi** [comment]
I can confirm:
Server: 20.04 Ubuntu Xorg
Client: Arch linux Gnome 3.36.1 Wayland

Connection is handled correctly, and scrolling on client in open window on client works. However the mouse pointer on the client is not moved.

```
[2020-05-07T10:58:35] INFO: switch from "serv" to "mtop" at 1919,787
[2020-05-07T10:58:35] INFO: leaving screen
[2020-05-07T10:58:35] INFO: screen "serv" updated clipboard 1
[2020-05-07T10:58:40] INFO: switch from "mtop" to "serv" at 69,1328
[2020-05-07T10:58:40] INFO: entering screen
```

**Comment by andybiss** [comment]
I can confirm as well:
Server: 20.04 Ubuntu Desktop (Xorg)
Client: Fedora Workstation 32 (Xorg, installed via flatpak)

connects and enters screen but no mouse or keyboard

**Comment by marzukia** [comment]
In the same boat, both machines Fedora32 running x11. Connects and logs reflect correct screen detection but both keyboard and mouse don't work

**Comment by simons-public** [comment]
For any of the people using Wayland in this thread (i.e. RHEL 8 uses Wayland by default now) [Wayland isn't supported by Barrier](https://github.com/debauchee/barrier/projects/1) as of right now.

**Comment by CurtisDyer** [comment]
So when I had this issue, it was actually user error. I had pressed scroll lock at some point, which locked the cursor to my server's screen. Just a heads-up for anyone else who's spacing out over the obvious (or maybe it's just me).

**Comment by p12tic** [comment]
Closing as a duplicate of #247. The bug for actual Wayland support is in #109.

**Comment by frohro** [comment]
This has been plaguing me for a good while.  I have a touch screen, and found that touching the touch screen allowed me to get into the other screen.  Here is the debug messages I was getting:
[2020-09-29T15:43:36] DEBUG1: try to leave "frohro-260" on right
[2020-09-29T15:43:36] DEBUG: locked by mouse buttonID: 0

**Comment by nautix** [comment]
This *just* started with me, too.

For months, I used a Debian Buster server and a Windows 10 client with Barrier 2.3.2 on both. Then the Debian Buster server went away and I booted up a little used machine running Peppermint Bionic. I reconfigured the Windows machine as the server and the Peppermint machine as the client. It worked great for a couple of days.

This morning, I ran a long-overdue 'apt-get upgrade' on the Peppermint box. After rebooting, Barrier still seemed to work fine, but, this afternoon, Barrier logs show a connection but the mouse pointer isn't moving to the client.

I poked around a bit, turning off various options and reloading, all to no avail. Then I noticed that the Peppermint box had a newer version of Barrier running: 2.3.3. I uninstalled Barrier 2.3.2 on the Windows server, then downloaded and installed Barrier 2.3.3. But no joy.

Reloading both client and server: still no mouse pointer on the client (except using the native touchpad).
Stopping and restarting both: no joy.
Quitting and restarting both: no joy.
Rebooting both: no joy.

The logs always show a successful connection at both ends, but the mouse pointer never moves on the client. (Yes, I made sure the server config had the client positioned in the correct location.)

*sigh*

Any (new) ideas?

**Comment by mic-lachmann** [comment]
I have the same problem between win10 and OSX. When I assign switch to max screen to a hotkey, it works well, and then I can move my mouse back to win10, but from win10 I can't go to the max screnn

**Comment by Motazzhd** [comment]
ok it was working fine with me when it was mac the server for 1 win and 1 other mac both as clients , however i wanted to use the win as server other 2 macs clients, everything connects well but the mouse doesn't travel.

**Comment by Motazzhd** [comment]
ok it was working fine with me when it was mac the server for 1 win and 1 other mac both as clients , however i wanted to use the win as server other 2 macs clients, everything connects well but the mouse doesn't travel.

**Comment by rajeshdgsb** [comment]
I had this same issue with barrier across a windos10 and Ubuntu20.04 .
Thanks to @CurtisDyer , I found that this was easy to fix by just undoing the scroll lock !

**Comment by Leathersoup** [comment]
I experience this issue from time to time.  I believe it is linked to running full screen apps on (in my case the barrier server). When the full screen application is shut down barrier does not seem to recover access to the screen edges.

**Comment by gogoprog** [comment]
Experiencing the same issue from Linux to OSX.
It was working fine then I had a connection problem. When this problem was solved the client can connect but the cursor cannot enter to the client.

**Comment by ilakast** [comment]
The accepted answer to https://askubuntu.com/questions/961304/how-do-you-switch-from-wayland-back-to-xorg-in-ubuntu-17-10 solved the issue for me. Thanks to @geertdobbels for the pointer

**Comment by COUPON4Vitamins** [comment]
I had the same issue. The culprit was "Scroll Lock." I didn't notice I had it on as the LED lights on my keyboard stopped working. Once I pressed on ScrLK, everything worked flawlessly again.

**Comment by frohro** [comment]
I have discovered that this happens when something in the first screen is not completed before I try to go to the second screen.  For example, if I open a menu, but don't select any item, but then try to go to the second screen, it will not allow me to do go there.

**Comment by andreluiztorres** [comment]
I've found it works fine when we're running a full-screen program. Minimized doesn't really allow you to hover from one screen to another....

**Comment by msilmser** [comment]
scroll lock was the answer for me

**Comment by henriquemod** [comment]
> 
> 
> I had the same issue. The culprit was "Scroll Lock." I didn't notice I had it on as the LED lights on my keyboard stopped working. Once I pressed on ScrLK, everything worked flawlessly again.



> 
> 
> scroll lock was the answer for me

Same here

**Comment by noob-max-ai** [comment]
> After starting barrier from the command shell, I saw a message related to a missing wayland-egl plugin, which made me think that wayland might be the problem (wayland is on by default in gnome 3.4 or at least in fedora31). This made me think that on my previous fedora31 install, I used to login under Xorg gnome, and not under wayland ..... So I logged out and logged back in but under xOrg gnome instead of the default wayland, and now barrier seems to be working. The question now is how to make barrier working under wayland....

Any updates?

**Comment by AymenFJA** [comment]
> I had the same problem with a Win7 Server and raspian clients. Scroll lock was turned on and that locked the mouse to the windows screens.

worked for me: I did not have a scroll lock in my physical keyboard but somehow it locked itself: https://support.microsoft.com/en-us/office/turn-off-scroll-lock-a78492cc-c124-4782-a38b-df46669520e0

**Comment by khaledbnmohamed** [comment]
> I had the same problem with a Win7 Server and raspian clients. Scroll lock was turned on and that locked the mouse to the windows screens.

Maaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaan, That is supeeeer stupidddd that I lost multiple hours in it

**Comment by Lucatronlk** [comment]
> I had this same issue with barrier across a windos10 and Ubuntu20.04. Thanks to @CurtisDyer, I found that this was easy to fix by just undoing the scroll lock!

This fixed my issue!! Thanks!

I have one MacBook left - Windows - right .. and suddenly it stopped working. I could not leave the screen. I did not have SSL activated, so when I activated SSL the issue persisted. And a simple scroll unlock fixed it 💯

**Comment by vklepper** [comment]
I had a similar problem after setting Windows display to 125%. Just in case this helps anyone.

Server: Win 11
Client: Ubuntu 22.04

**Comment by owolowo955** [comment]
I had the same problem, switching my Ubuntu 22.04 (client) from Wayland to Xorg Gnome. 
This fixed my issue!! Thanks!

**Comment by deveciabdullah** [comment]
vi /etc/gdm3/custom.conf
#WaylandEnable=false ###uncomment this line for disable wayland

systemctl restart gdm3

This fixed my issue

**Comment by DJArty** [comment]
> nano /etc/gdm3/custom.conf 
>WaylandEnable=false ###uncomment this line for disable wayland
> 
> systemctl restart gdm3

Working for me too, but dont think its normal to disable Wayland for all system just for barrier can work. Must work with wayland (not working now)
barrier from snap:
barrier                    2.4.0-4-gac5a1bfd  682    latest/stable    shymega  
Server: Ubuntu 22.04 (Wayland by default)
Client: Windows 10

**Comment by architdbz** [comment]
I had exactly same problem when I make Ubuntu as my server. When I make windows as my server the mouse is able to travel across the screen but becomes invisible. I found the solution to this problem in this link.

https://askubuntu.com/questions/1407275/mouse-disappears-while-using-barrier

All you have to do is edit custom.conf

$ sudo nano /etc/gdm3/custom.conf

uncomment this line if its already commented, and make it false 
"WaylandEnable=false"

save and exit and then type
$ sudo systemctl restart gdm3

Barrier should now work properly. Hope this helps.

**Comment by stephenskett** [comment]
> vi /etc/gdm3/custom.conf #WaylandEnable=false ###uncomment this line for disable wayland
> 
> systemctl restart gdm3
> 
> This fixed my issue

Same here.
Weirdly, manually selecting the 'Ubuntu with xOrg' option at login screen didn't help in my case, but disabling Wayland in the GDM3 config did. 😖 

Here's my setup, FWIW...
Server: Windows 11
Client: Ubuntu 22.04 LTS

**Comment by YuriyL-Git** [comment]
For me helped   to set Fix Scroll checkbox in settings

**Comment by Smartog** [comment]
> > I had the same problem with a Win7 Server and raspian clients. Scroll lock was turned on and that locked the mouse to the windows screens.
> 
> worked for me: I did not have a scroll lock in my physical keyboard but somehow it locked itself: https://support.microsoft.com/en-us/office/turn-off-scroll-lock-a78492cc-c124-4782-a38b-df46669520e0

yes！yes！this way works for me. After pressing the SrcLk key on my keyboard, issue fixed!

**Comment by cieske** [comment]
Can't sharing mouse suddenly happens, disabling scroll lock is the answer for me too:)

**Comment by AhmedKhan-GH** [comment]
> I had exactly same problem when I make Ubuntu as my server. When I make windows as my server the mouse is able to travel across the screen but becomes invisible. I found the solution to this problem in this link.
> 
> https://askubuntu.com/questions/1407275/mouse-disappears-while-using-barrier
> 
> All you have to do is edit custom.conf
> 
> $ sudo nano /etc/gdm3/custom.conf
> 
> uncomment this line if its already commented, and make it false "WaylandEnable=false"
> 
> save and exit and then type $ sudo systemctl restart gdm3
> 
> Barrier should now work properly. Hope this helps.

You are a legend

**Comment by Dakkaron** [comment]
Late to the party, but switching over to [Input Leap](https://github.com/input-leap/input-leap) fixed the issue for me. They just released v3.0.0, which adds Wayland support.

(From what I gather, Barrier is discontinued and the devs forked it and renamed it Input Leap. Input Leap seems to be compatible to Barrier, so I'm currently running Barrier on one device and Input Leap on the other and it works. Don't know how long that will be the case, so update to Input Leap if you can.)

**Comment by frohro** [comment]
Wow!  Thanks for passing on the good news!
